test/numbers/numbers.api-test.ts

```diff
@@ -0,0 +1,229 @@
+      },
+    ];
+    const { status, body } = await api.post(payload).to('/numbers');
+    expect(status).toEqual(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toContain('createdBy must be shorter than or equal to 60 characters');
+    expect(body.message).toContain('createdBy must be shorter than or equal to 60 characters');
+  });
+
+  it(`POST /numbers single, missing fields`, async () => {
+    const payload = [{}];
+    const { status, body } = await api.post(payload).to('/numbers');
+    expect(status).toEqual(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toContain('numberTypeId should not be empty');
+    expect(body.message).toContain('createdBy should not be empty');
+    expect(body.message).toContain('requestingSystem should not be empty');
+  });
+
+  it(`POST /numbers single, empty payload`, async () => {
+    const payload = '';
+    const { status, body } = await api.post(payload).to('/numbers');
+    expect(status).toEqual(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toMatch('Validation failed (parsable array expected)');
+  });
+
+  it(`POST /numbers single, empty array`, async () => {
+    const payload = [];
+    const { status, body } = await api.post(payload).to('/numbers');
+    expect(status).toEqual(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toMatch('Request payload is empty');
+  });
+
+  it(`POST /numbers single, empty array`, async () => {
+    const payload = '[]';
+    const { status, body } = await api.post(payload).to('/numbers');
+    expect(status).toEqual(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toMatch('Validation failed (parsable array expected)');
+  });
+
+  it(`POST /numbers single, bad json`, async () => {
+    const payload = 'asd';
+    const { status, body } = await api.post(payload).to('/numbers');
+    expect(status).toEqual(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toMatch('Validation failed (parsable array expected)');
+  });
+
+  it(`POST /numbers multiple numbers`, async () => {
+    const payload = [
+      {
+        numberTypeId: 1,
+        createdBy: 'Jest',
+        requestingSystem: 'Jest 1 - Deal',
+      },
+      {
+        numberTypeId: 1,
+        createdBy: 'Jest',
+        requestingSystem: 'Jest 2 - Facility',
+      },
+      {
+        numberTypeId: 2,
+        createdBy: 'Jest',
+        requestingSystem: 'Jest 3 - Party',
+      },
+      {
+        numberTypeId: 8,
+        createdBy: 'Jest',
+        requestingSystem: 'Jest 4 - Covenant',
+      },
+    ];
+    const { status, body } = await api.post(payload).to('/numbers');
+    expect(status).toEqual(201);
+    expect(body).toHaveLength(4);
+
+    /* eslint-disable security/detect-object-injection */
+    for (let i = 0; i++; i < 4) {
```

Comment:
I wonder if `test.each()` will be more appropriate here?
> https://jestjs.io/docs/api#testeachtablename-fn-timeout

Reply:
Will change in next PR